r-aks.tf

```diff
@@ -35,6 +35,13 @@ resource "azurerm_kubernetes_cluster" "aks" {
     identity_ids = [azurerm_user_assigned_identity.aks_user_assigned_identity.id]
   }
 
+  dynamic "auto_scaler_profile" {
+    for_each = var.auto_scaler_expander_name != null ? [true] : []
+    content {
+      expander = var.auto_scaler_expander_name
+    }
+  }
+
```

Comment:
This must be a block with all `auto_scaler_profile` parameters. See suggestion on the variable.

Reply:
Added, tested
Seems to be working 
Thanks for the suggestion